### PR TITLE
cabal init -i: add the GHC2021 language option

### LINKED ISSUE #8265: cabal init: suggest GHC2021?
`cabal init -i` could suggest `GHC2021` along with `Haskell*` perhaps? I'm thinking that in the interactive menu we could write `"GHC2021 (requires at least GHC 9.2)"`. The language is already supported by Cabal
https://github.com/haskell/cabal/blob/f1395303feba296c58ba7eea3ecafdc10df7468b/Cabal-syntax/src/Language/Haskell/Extension.hs#L60
and it's just a matter of giving the extension more visibility. 

The non-interactive mode already has a flag that can be set to whatever (unfortunately, there's no check that the value makes sense), so it doesn't need any change.

Thoughts?

### PR DESCRIPTION
**Pull request by ulysses4ever**
fix  #8265
---
Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [x] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [ ] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: cabal-install/src/Distribution/Client/Init/Interactive/Command.hs [hunk 1]
```diff
@@ -413,17 +413,22 @@ testDirsPrompt flags = getTestDirs flags $ do
...
-      "Haskell2010" -> return Haskell2010
-      "Haskell98" -> return Haskell98
-      l | all isAlphaNum l -> return $ UnknownLanguage l
-      _ -> do
+    if
+      | l == h2010       -> return Haskell2010
+      | l == h98         -> return Haskell98
+      | l == ghc2021     -> return GHC2021
+      | all isAlphaNum l -> return $ UnknownLanguage l
+      | otherwise        -> do
```
Comment (ffaf1) [inline cabal-install/src/Distribution/Client/Init/Interactive/Command.hs:431]: Not related to this PR but: weird UX choice. If I mistype I would like “Unknown language, possible choices are α, β, γ” printed to screen; I do not understand what “language must be alphanumeric” is useful for.

**Review by ffaf1 — APPROVED** [review]

**Comment by jneira** [comment]
good work, I would say this is in the limit of needing tests, maybe an unit test?

File: cabal-install/src/Distribution/Client/Init/Interactive/Command.hs (hunk 1 above)
Comment (ulysses4ever) [reply cabal-install/src/Distribution/Client/Init/Interactive/Command.hs:431]: Sounds reasonable. Perhaps, opening a ticket is a way forward.

**Comment by ulysses4ever** [comment]
@jneira there's a test that the new language is on the list:
https://github.com/haskell/cabal/pull/8277/files#diff-087fb21b3293d2ef32df883500fad0ead4fa4281b0ba0d376ea36928164567afR915

What else can be tested here?

**Comment by jneira** [comment]
hmm, maybe not an unit test but a test checking the produced cabal file has the correct language set, but I hope it is already tested with other values somewhere

**Review by jneira — APPROVED** [review]
lgtm, thanks again!

File: cabal-install/src/Distribution/Client/Init/Interactive/Command.hs (hunk 1 above)
Comment (Mikolaj) [reply cabal-install/src/Distribution/Client/Init/Interactive/Command.hs:431]: the ticket: #8278

**Comment by ulysses4ever** [comment]
@jneira thanks! I went ahead and added a test that [actually checks the contents of the cabal file](https://github.com/haskell/cabal/pull/8277/files#diff-6dc1ee73e151eb71a44c6d12783bc320f4016f26caab83f541185e9e19a57beaR11).

**Comment by ulysses4ever** [comment]
@Mergify backport 3.8

**Pull request merged**